src/Program.cs

```diff
@@ -149,7 +138,7 @@ private static async Task<UpstreamResponse> SendUpstreamRequest(HttpRequest requ
                     }
                 }
 
-                foreach (var header in request.Headers.Where(h => !_excludedRequestHeaders.Contains(h.Key) && !_contentRequestHeaders.Contains(h.Key)))
+                foreach (var header in request.Headers.Where(h => !_excludedRequestHeaders.Contains(h.Key) && !_contentRequestHeaders.Contains(h.Key) && !h.Key.Equals(_responseSelectionHeader)))
```

Comment:
Instead I think you should add `_responseSelectionHeader` to `_excludedRequestHeaders`.